File: src/main/java/ir/syrent/wanted/Listeners/PlayerQuitListener.java
package ir.syrent.wanted.Events;

import ir.syrent.wanted.Utils.SkullBuilder;
import org.bukkit.entity.Player;
import org.bukkit.event.EventHandler;
import org.bukkit.event.Listener;
import org.bukkit.event.player.PlayerQuitEvent;

public class PlayerQuitListener implements Listener {

    @EventHandler
    public void onQuit(PlayerQuitEvent event) {
        Player player = event.getPlayer();
        //Removing player from cache to avoid memory leaks
        SkullBuilder.getInstance().cache.remove(player);
    }

}

File: src/main/kotlin/ir/syrent/wanted/listener/PlayerQuitListener.kt
package ir.syrent.wanted.listener

import ir.syrent.wanted.core.WantedManager
import ir.syrent.wanted.storage.Database
import me.mohamad82.ruom.Ruom
import org.bukkit.event.EventHandler
import org.bukkit.event.Listener
import org.bukkit.event.player.PlayerQuitEvent

class PlayerQuitListener: Listener {

    init {
        Ruom.registerListener(this)
    }

    @EventHandler
    fun onPlayerQuit(event: PlayerQuitEvent) {
        val player = event.player

        if (WantedManager.wPlayers.containsKey(player.uniqueId)) {
            Database.instance.saveWPlayer(WantedManager.getWPlayer(player))
        }
    }
}